piqe_ocp_lib/tests/resources/test_ocp_resource_quotas.py

```diff
@@ -0,0 +1,162 @@
+        logger.info(f"Create an object count V1ResourceQuotaSpec")
+        rq_item = ocp_rq.resource_quota_spec().persistent_volume_claims(count="8").build()
+        logger.info(f"Create Response : {rq_item}")
+        if not rq_item.hard.get('persistentvolumeclaims') == '8':
+            assert False, f"Failed to create object count V1ResourceQuotaSpec"
+
+    def test_storage_resource_quota_spec_builder(self, ocp_rq):
+        logger.info(f"Create a storage V1ResourceQuotaSpec")
+        rq_item = ocp_rq.resource_quota_spec().ephemeral_storage(requests="1G").\
+            storage_class(class_name='gold', requests='8G').build()
+        logger.info(f"Create Response : {rq_item}")
+        if not rq_item.hard.get('requests.ephemeral-storage') == '1G' and \
+                not rq_item.hard.get('gold.storageclass.storage.k8s.io/requests.storage') == '8G':
+            assert False, f"Failed to create storage V1ResourceQuotaSpec"
+
+    def test_selector_expression_resource_quota_spec_builder(self, ocp_rq):
+        logger.info(f"Create a PriorityClass V1ScopedResourceSelectorRequirement")
+        rq_item = ocp_rq.resource_quota_spec().scope_selector().priority_class_scope()\
+            .in_().values(vals=['high']).done().build()
+        logger.info(f"Create Response : {rq_item}")
+        if not rq_item.scope_selector.match_expressions[-1].operator == 'In' and \
+                len(rq_item.scope_selector.match_expressions[-1].values) != 1:
+            assert False, f"Failed to create a priority class V1ResourceQuotaSpec"
+
+    def test_selector_expression_resource_quota_spec_rule_check(self, ocp_rq):
+        logger.info(f"Create a V1ScopedResourceSelectorRequirement")
+        with pytest.raises(Exception):
+            ocp_rq.resource_quota_spec().scope_selector().terminating_scope().not_exists().done().build()
+
+    def test_extended_resource_quota_spec_builder(self, ocp_rq):
+        logger.info(f"Create an extended resource V1ResourceQuotaSpec")
+        rq_item = ocp_rq.resource_quota_spec().resource(resource="nvidia.com/gpu", requests="4").build()
+        logger.info(f"Create Response : {rq_item}")
+        if not 'requests.nvidia.com/gpu' in rq_item.hard and \
+                not rq_item.hard.get('requests.nvidia.com/gpu') == '4':
+            assert False, f"Failed to create extended resource V1ResourceQuotaSpec"
+
+    def test_extended_resource_count_spec_builder(self, ocp_rq):
+        logger.info(f"Create a standard resource count V1ResourceQuotaSpec")
+        rq_item = ocp_rq.resource_quota_spec().resource(resource="deployments.apps", count="4").build()
+        logger.info(f"Create Response : {rq_item}")
+        if not 'count/deployments.apps' in rq_item.hard and \
+                not rq_item.hard.get('count/deployments.apps') == '4':
+            assert False, f"Failed to create standard resource count V1ResourceQuotaSpec"
+
+    def test_create_resource_quota(self, ocp_rq):
```

Comment:
Since the cleanup of creating this rq is achieved by the test_delete_resource_quota, this would mean that to ensure that the test environment is brought back to its original baseline that the minimum level of granularity is at the test module level. Other tests in this directory are done at the method level. For the sake of consistency, we should agree on what scope we want to agree to. I can think of pros and cons to each, so it's a matter of what we collectively want to stick to. Hope this makes sense.

Reply:
@btjd I agree with that as well. Whatever we decide with, I would need to make the same changes for the LimitRange API (in a separate PR) since I used a particular Ocp Resource class as a template(I think OcpConfigMap) for both the LimitRange and ResourceQuota Resource APIs.

Reply:
I think we should bring up this topic in our next meeting so we can come up with a decision. So far we have handled this at the method level. That however can be costly in terms of execution times for some tests to a point where it's hard to justify it.